048-avoid-broker-restarts-when-in-recovery.md

```diff
@@ -65,24 +69,19 @@ The proposed flow of KafkaRoller
 
-As described in the graph above, the following steps are added before the retry step:
+As described in the graph above, the following steps are added after checking the connection to the broker:
 
-1. Ensure that the Kafka pod is not in a stuck state. The existing `isPodStuck()` can be used for this. If the pod is stuck, continue with the current behaviour to retry rolling of the pod.
-2. Use the Pod DNS name to connect to the KafkaAgent API endpoint. If it cannot connect to the endpoint, continue with the current behaviour to retry rolling of the pod. (The connection might timeout or refused because KafkaAgent is stuck or running older version).
+1. If the AdminClient connection failed, check the broker state instead of force restarting the pod.
+2. Use the Pod DNS name to connect to the KafkaAgent API endpoint. If it cannot connect to the endpoint (the connection might timeout or refused because KafkaAgent is stuck or running older version), continue with the current behaviour.
 3. Use the cluster operator's client certificate and keys for mTLS authentication.
 4. Send `GET v1/broker-state` request to the endpoint.
-5. If the HTTP return code is not 200, continue with the current behaviour to retry rolling of the pod.
-6. If the `brokerState` value is `2` (`RECOVERY`), follow the new logic for handling log recovery. Otherwise continue with the current behaviour to retry rolling of the pod.
-
-*Handling log recovery*
-
-7. Set `inLogRecovery` to true for the pod  (`inLogRecovery` is a new boolean that would be added to the `RestartContext`).
-8. Log the values of `remainingLogsToRecover` and `remainingSegmentsToRecover` from the JSON response in an `INFO` message.
-9. Continue with the current behaviour to retry rolling of the pod.
-10. On the next retry, check `inLogRecovery` inside the `restartIfNecessary()` function.
-11. When `inLogRecovery` is set to true, call `awaitReadiness()` which periodically checks and waits for readiness until the operational timeout reaches.
-13. If the operational timeout is reached, repeat from step1.
-14. If the maximum number of attempts has been reached, include log recovery in the reason for failure.
+5. If the HTTP return code is not 200, continue with the current behaviour.
+6. If the `brokerState` value is `2` (`RECOVERY`)
+     - Log the values of `remainingLogsToRecover` and `remainingSegmentsToRecover` from the JSON response in an `INFO` message.
+     - Call `awaitReadiness()` which periodically checks and waits for readiness until the operational timeout reaches.
+7.  If the `brokerState` value is not `2` (`RECOVERY`), continue with the current behaviour of force restarting the pod and waiting for readiness until the operational timeout reaches.
+8. If the operational timeout is reached, continue with the current behaviour of rescheduling the rolling of the pod. This will result in repeating from step 1 if broker is still in log recovery.
```

Comment:
```suggestion
8. If the operational timeout is reached, continue with the current behaviour of rescheduling the rolling of the pod. This will result in repeating from step 1 if the broker is still in log recovery.
```